app/models/folio/instance.rb

```diff
@@ -14,6 +14,46 @@ def self.fetch(request, _live_lookup = true)
 
       Folio::Instance.from_dynamic(data)
     end
+
+    def self.filter_parent_bound_withs(parent_bound_withs, hrid)
+      return [] unless parent_bound_withs
+
+      parent_bound_withs.flatten.compact.each do |bound_with_record|
+        @filter_parent_bound_withs ||= bound_with_record.dig('instance', 'items').each.map do |item|
+          boundwithhrids = item['boundWithHoldingsPerItem'].filter { |bw| bw.dig('instance', 'hrid') == hrid }.flatten
+          update_item_fields(item, boundwithhrids.first) if boundwithhrids.present?
+        end
+      end
+      @filter_parent_bound_withs.compact
```

Comment:
```suggestion
      parent_bound_withs.flatten.compact.each do |bound_with_record|
        filter_parent_bound_withs ||= bound_with_record.dig('instance', 'items').each.map do |item|
          boundwithhrids = item['boundWithHoldingsPerItem'].filter { |bw| bw.dig('instance', 'hrid') == hrid }.flatten
          update_item_fields(item, boundwithhrids.first) if boundwithhrids.present?
        end
      end
      filter_parent_bound_withs.compact
```

I don't think we want to memoize these at the class level here.

Reply:
@cbeer It breaks the application if we remove the memoization.